govcd/nsxt_application_profile_test.go

```diff
@@ -0,0 +1,182 @@
+	. "gopkg.in/check.v1"
+)
+
+func (vcd *TestVCD) Test_NsxtApplicationPortProfileProvider(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointAppPortProfiles)
+
+	appPortProfileConfig := getAppProfileProvider(vcd, check)
+	testAppPortProfile(appPortProfileConfig, types.ApplicationPortProfileScopeProvider, vcd, check)
+
+}
+
+func (vcd *TestVCD) Test_NsxtApplicationPortProfileTenant(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointAppPortProfiles)
+
+	appPortProfileConfig := getAppProfileTenant(vcd, check)
+	testAppPortProfile(appPortProfileConfig, types.ApplicationPortProfileScopeTenant, vcd, check)
+
+}
+
+func (vcd *TestVCD) Test_NsxtApplicationPortProfileReadSystem(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointAppPortProfiles)
+
+	testApplicationProfilesForScope(types.ApplicationPortProfileScopeSystem, check, vcd)
+}
+
+func (vcd *TestVCD) Test_NsxtApplicationPortProfileReadProvider(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointAppPortProfiles)
+
+	testApplicationProfilesForScope(types.ApplicationPortProfileScopeProvider, check, vcd)
+}
+
+func (vcd *TestVCD) Test_NsxtApplicationPortProfileReadTenant(check *C) {
+	skipNoNsxtConfiguration(vcd, check)
+	skipOpenApiEndpointTest(vcd, check, types.OpenApiPathVersion1_0_0+types.OpenApiEndpointAppPortProfiles)
+
+	testApplicationProfilesForScope(types.ApplicationPortProfileScopeTenant, check, vcd)
+}
+
+func getAppProfileProvider(vcd *TestVCD, check *C) *types.NsxtAppPortProfile {
+	nsxtManager, err := vcd.client.QueryNsxtManagerByName(vcd.config.VCD.Nsxt.Manager)
+	check.Assert(err, IsNil)
+
+	nsxtManagerUuid, err := GetUuidFromHref(nsxtManager[0].HREF, true)
+	check.Assert(err, IsNil)
+
+	nsxtManagerUrn, err := BuildUrnWithUuid("urn:vcloud:nsxtmanager:", nsxtManagerUuid)
+	check.Assert(err, IsNil)
+
+	// For PROVIDER scope application port profile must have ContextEntityId set as NSX-T Managers URN and no Org
+	appPortProfileConfig := &types.NsxtAppPortProfile{
+		Name:        check.TestName() + "PROVIDER",
+		Description: "Provider config",
+		ApplicationPorts: []types.NsxtAppPortProfilePort{
+			types.NsxtAppPortProfilePort{
+				Protocol:         "TCP",
+				DestinationPorts: []string{"11000-12000"},
+			},
+		},
+		ContextEntityId: nsxtManagerUrn,
+		Scope:           types.ApplicationPortProfileScopeProvider,
+	}
+	return appPortProfileConfig
+}
+
+func getAppProfileTenant(vcd *TestVCD, check *C) *types.NsxtAppPortProfile {
+	org, err := vcd.client.GetOrgByName(vcd.config.VCD.Org)
+	check.Assert(err, IsNil)
+
+	// For PROVIDER scope application port profile must have ContextEntityId set as NSX-T Managers URN and no Org
```

Comment:
Just checking is the comment correct? The function seems to be for tenant.
Few lines below Description is also referring to Provider.

BTW, where does the all-capitalized form of "PROVIDER" and "TENANT" come from? Looks to be "yelling" - not sure if that's really needed here :)

Reply:
I probably have overused capitalization :). These are `scope`values being set in API.

Basically there are 3 scopes `system`, `provider` and `tenant`

* `system` - read-only (built-in application profiles that are provided by `system` scope)
* `provider` - can be created and filtered by System user on a particular network provider (NSX-T manager). This means that whatever is set - should be available to all tenants using the same NSX-T manager
* `tenant` can be created at tenant level (org VDC)

Reply:
Please make a decision on this together with @dataclouder

Reply:
If the API demands uppercase, I prefer keeping the original value in the comments and docs